packages/amplify_api/lib/graphql/graphql_helpers.dart

```diff
@@ -0,0 +1,131 @@
+      default:
+        return "Error: UnknownType!";
+    }
+  }
+
+  List<String> _getFieldsFromModelType(Map<String, ModelField?>? fieldsMap) {
+    List<String> fieldsList = [];
+    if (fieldsMap != null) {
+      fieldsMap.forEach((key, value) {
+        if (value?.association == null) fieldsList.add(key);
+      });
+    }
+    return fieldsList;
+  }
+
+  ModelSchema _getSchema(ModelType modelType) {
+    ModelProviderInterface? provider = AmplifyAPI.instance.getModelProvider();
+
+    if (provider == null) {
+      throw ApiException('No modelProvider found',
+          recoverySuggestion:
+              'Pass in a modelProvider instance while instantiating APIPlugin');
+    }
+
+    ModelSchema schema = provider.modelSchemas.firstWhere(
+        (elem) => provider.getModelTypeByModelName(elem.name) == modelType);
+
+    if (schema == null) {
+      throw ApiException('No schema found for the ModelType provided',
+          recoverySuggestion:
+              'Pass in a valid modelProvider instance while instantiating APIPlugin or provide a vaild ModelType');
+    }
+
+    return schema;
+  }
+
+  @override
+  GraphQLRequest<T> buildQuery<T extends Model>(
+      {required ModelType modelType,
+      required Map<String, String>? variableInput,
+      required String? id,
+      required GraphQLRequestType requestType,
+      required GraphQLRequestOperation requestOperation}) {
+    ModelSchema schema = _getSchema(modelType);
+
+    String? modelName = requestOperation == GraphQLRequestOperation.list
+        ? schema.pluralName
+        : schema.name;
+    String name = modelName ?? 'No model name found';
+    String fields = _getFieldsFromModelType(schema.fields).join(' ');
+
+    String reqTypeStr = describeEnum(requestType.toString());
+    String reqOperationStr = describeEnum(requestOperation.toString());
+
+    var varInputUpperStr = '';
+    var varInputLowerStr = '';
+
+    if (variableInput!.isNotEmpty) {
+      List<String> variableInputUpper = [];
+      List<String> variableInputLower = [];
+
+      for (var key in variableInput.keys) {
+        String? modelInputType = variableInput[key];
+        variableInputUpper.add("\$$key: $modelInputType!");
+        variableInputLower.add("$key: \$$key");
+      }
+
+      varInputUpperStr = "(${variableInputUpper.join(", ")})";
+      if (requestOperation == GraphQLRequestOperation.get) {
+        varInputLowerStr = "(${variableInputLower.join(", ")})";
+      } else {
+        varInputLowerStr = "(input: { ${variableInputLower.join(", ")} })";
+      }
+    }
+
+    if (requestOperation == GraphQLRequestOperation.list) {
+      fields = 'items { $fields } nextToken';
+    }
+
+    String doc =
+        '''$reqTypeStr $reqOperationStr$name${varInputUpperStr} { $reqOperationStr$name$varInputLowerStr { $fields } }''';
+
+    // TODO: create input map for variables & connect with current variableInput
+    Map<String, dynamic> variables = id != null ? {"id": id} : const {};
```

Comment:
`GraphQLRequest` already has `{}` defined as default param in constructor, so I would not suggest repeating that default here. Instead, could `variables` just be null when no id so default constructor works as is?

Reply:
Since variables is being passed in regardless of value on the next line, the constructor won't accept a `null` value. It has to be at least be an empty Map. I did remove the unneeded `const`.